app/controllers/homepage_controller.rb

```diff
@@ -21,7 +21,9 @@ def index
     listing_shape_menu_enabled = all_shapes.size > 1
     @show_categories = @categories.size > 1
     show_price_filter = @current_community.show_price_filter && all_shapes.any? { |s| s[:price_enabled] }
-    @show_custom_fields = @current_community.custom_fields.any?(&:can_filter?) || show_price_filter
+
+    filters = @current_community.custom_fields.where(show_filter: true).sort
```

Comment:
This needs at the minimum a DB index for the query that is generated. It would also feel like good idea to cache this to memcached and/or in-memory cache. There are loaded on every single pageload to homepage.